src/main/java/firebase/FirebaseInitializer.java

```diff
@@ -24,8 +28,6 @@ public static Firestore initializeFirebase() {
             System.err.println("Error initializing Firebase: " + e.getMessage());
             return null;
         }
-
-        return initializeFirebase();
```

Comment:
Its good that you removed the extra  return statement, that would never be reached